### PR TITLE
feat: subscribe to owner programs on delegation

### PR DESCRIPTION
**Pull request by thlorenz**
## Summary

Implements program subscription on delegation to capture undelegation updates with resilience.
When delegated accounts are cloned or an account subscription received in which an account was
delegated, the validator now subscribes to the original owner
program to detect when undelegation occurs, enabling more reliable state tracking.

CLOSES: #819

## Details

When an account is delegated we now subscribe to the original owner program for account updates.
This allows us to receive program-level account updates that include undelegation events, providing resilience against missing state changes.

Includes unit tests for all cases.

## Summary by CodeRabbit

* **Bug Fixes**
  * When processing delegated accounts, the system now also triggers best-effort background subscriptions to original owner programs (concurrent, non-blocking; failures are logged).
  * Prevents re-creating subscriptions for programs already subscribed, reducing redundant work.

* **Tests**
  * Added extensive tests for program-subscription behavior covering delegation, concurrency, deduplication, race conditions, and edge cases.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: magicblock-chainlink/src/chainlink/fetch_cloner/mod.rs [hunk 1]
```diff
@@ -368,8 +368,23 @@ where
...
+                                        .subscribe_program(
+                                            delegation_record.owner,
+                                        )
+                                        .await
+                                    {
+                                        warn!(
+                                            "Failed to subscribe to owner program {} for account {}: {}",
+                                            delegation_record.owner, pubkey, err
+                                        );
+                                    }
```
[the remark this answers is not in the document]
Comment (thlorenz) [reply magicblock-chainlink/src/chainlink/fetch_cloner/mod.rs]: Fixed in b6deb162
